src/WCSimConstructMaterials.cc

```diff
@@ -25,7 +27,9 @@ void WCSimDetectorConstruction::ConstructMaterials()
 
 
   G4NistManager *nist_man = G4NistManager::Instance();
-
+  
+  //---BGO
+  G4Material* BGO = nist_man->FindOrBuildMaterial("G4_BGO");
```

Comment:
```suggestion
  BGO = nist_man->FindOrBuildMaterial("G4_BGO");
```
BGO is defined as a class variable. This means you can use it

Reply:
Thanks again! This solves the no-valid-material error. Sorry for the trouble. Sometimes I find confusing about when a variable is a class variable or not...

Anyway, with the BGO being defined in WCSimConstructMaterials.cc this way, the optical properties are not added correctly to the BGO, because I see no Scintillation...

Reply:
Can you add a printout inside 
```
if (placeBGOGeometry) {
```
in WCSimDetectorConstruction, to say something like "Placing AmBe source in geometry at x,y,z", to ensure that the calibration source is actually being built?

Reply:
Here it is, it prints correctly. I am also seeing the BGO correctly plotted in the "interactive mode".
```
Placing AmBe source in geometry at (0,0,0)
Geofile written
FINISHED WCSimDetectorConstruction::CONSTRUCT
```

Reply:
Hmm. You mean you see the BGO target in a geometry visualisation?

Were you seeing scintalliation light when the BGO material was defined in `WCSimDetectorConstruction.cc`? If you were, there's probably something missing in the transfer to `WCSimConstructMaterials.cc`

Reply:
If by "something missing in the transfer" you meant something was missing in the BGO MPT, you were right. Don't know what was missing but copied that lines from the previous WCSimDetectorConstruction into WCSimConstructMaterial and now I can see Scintillation! Thanks again Tom 👍